connectors/opa/connector.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"fybrik.io/fybrik/pkg/connectors/datacatalog/clients"
+	"fybrik.io/fybrik/pkg/model/datacatalog"
+	"fybrik.io/fybrik/pkg/model/policymanager"
+	"github.com/gin-gonic/gin"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+const (
+	headerCredentials = "X-Request-Cred"
+	policyEndpoint    = "/v1/data/dataapi/authz/verdict"
+)
+
+type ConnectorController struct {
+	OpaServerURL  string
+	OpaClient     *retryablehttp.Client
+	CatalogClient clients.DataCatalog
+}
+
+func NewConnectorController(opaServerURL string, catalogClient clients.DataCatalog) *ConnectorController {
+	return &ConnectorController{
+		OpaServerURL:  opaServerURL,
+		OpaClient:     retryablehttp.NewClient(),
+		CatalogClient: catalogClient,
+	}
+}
+
+func (r *ConnectorController) GetPoliciesDecisions(c *gin.Context) {
+	// Parse request
+	var request policymanager.GetPolicyDecisionsRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	// Get asset metadata from catalog connector
+	requestToCatalog := &datacatalog.GetAssetRequest{
+		AssetID:       request.Resource.ID,
+		OperationType: datacatalog.READ,
+	}
+	assetInfo, err := r.CatalogClient.GetAssetInfo(requestToCatalog, c.GetHeader(headerCredentials))
+	if err != nil {
+		// TODO: better error propagation
```

Comment:
Is this TODO for this PR or in the future?

Reply:
future